cmd/runtimetest/main.go

```diff
@@ -370,6 +371,62 @@ func validateMountsExist(spec *rspec.Spec) error {
 	return nil
 }
 
+func getActualCgroupPath() (string, error) {
+	f, err := os.Open("/proc/self/cgroup")
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+
+	cgroupPath := ""
+	s := bufio.NewScanner(f)
+	for s.Scan() {
+		if err := s.Err(); err != nil {
+			return "", err
+		}
+
+		text := s.Text()
+		parts := strings.Split(text, ":")
+		if cgroupPath == parts[2] && cgroupPath != "" {
+			continue
+		} else if cgroupPath == "" && parts[2] != "" {
+			cgroupPath = parts[2]
+		} else {
+			return "", fmt.Errorf("error path with cgroup controllers")
+		}
+	}
+
+	if cgroupPath == "" {
+		return "", fmt.Errorf("can not get cgroup path")
+	}
+	return cgroupPath, nil
+}
+
+func validateCgroupsPath(spec *rspec.Spec) error {
+	logrus.Debugf("validating cgroupsPath")
+	expectedPath := spec.Linux.CgroupsPath
+	if expectedPath == nil {
+		return nil
+	}
+	*expectedPath = strings.Replace(*expectedPath, ":", "/", -1)
+
+	actualPath, err := getActualCgroupPath()
+	if err != nil {
+		return err
+	}
+
+	if filepath.IsAbs(*expectedPath) {
+		if *expectedPath != actualPath {
+			return fmt.Errorf("Cgroup path expected: %v, actual: %v", *expectedPath, actualPath)
+		}
+	} else {
+		if _, err := filepath.Rel(*expectedPath, actualPath); err != nil {
```

Comment:
Basically, this branch should either be a noop or should create a new container and compare the value. Those are the only two things we can safely do here.